prereise/cli/download/tests/test_download_manager.py

```diff
@@ -0,0 +1,12 @@
+from unittest.mock import MagicMock, patch
+
+from prereise.cli.download.download_manager import generate_parser
+
+
+@patch("prereise.cli.download.download_manager.get_data_sources_list")
+@patch("prereise.cli.download.download_manager.add_data_source_to_download_parser")
+def test_generate_parser(add_data_source_to_download_parser, get_data_sources_list):
```

Comment:
I would consider leaving this one out - whether or not something is called should ideally be an implementation detail that a test shouldn't know about. Note, I'd consider this distinct from the examples in `test_data_sources.py` since those involve testing the effects of the function being called.

One thing that might be nice is to test the parser itself. Here is an example I put together, feel free to add it or make changes if you like.
```
def test_parser_wind_data():
    parser = generate_parser()
    start_date = "2020-01-01"
    end_date = "2021-01-01"

    arglist = [
        "wind_data",
        f"--start_date={start_date}",
        f"--end_date={end_date}",
        "--file_path=./data",
        "-r=Texas",
    ]
    result = parser.parse_args(arglist)
    assert start_date == result.start_date
    assert end_date == result.end_date

```

Reply:
nice suggestion, i just realized i don't really have many tests validating the actual parsing - was testing that mostly manually locally via terminal. Added tests for those and some 👍

Reply:
Nice, looks good!